src/Response/RedirectResponse.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Drupal\commerce\Response;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Routing\TrustedRedirectResponse;
+
+/**
+ * Provides an exception that represents the need for an HTTP redirect.
+ */
+class RedirectResponse extends \LogicException {
+
+  /**
+   * The URL to redirect to.
+   *
+   * @var string
+   */
+  protected $url;
+
+  /**
+   * The status code to use.
+   *
+   * @var int
+   */
+  protected $statusCode;
+
+  /**
+   * The headers to set.
+   *
+   * @var string[]
+   */
+  protected $headers;
+
+  /**
+   * Creates a new RedirectResponse object.
+   *
+   * @param string $url
+   *    The URL to redirect to.
+   * @param int $status_code
+   *    The redirect status code.
+   * @param string[] $headers
+   *    Headers to pass with the redirect.
+   */
+  public function __construct($url, $status_code = 302, array $headers = []) {
+    if (!UrlHelper::isValid($url)) {
+      throw new \InvalidArgumentException('Invalid URL for secure redirect');
+    }
+    $this->url = $url;
+    $this->statusCode = $status_code;
+    $this->headers = $headers;
+  }
+
+  /**
+   * Get the redirect URL.
+   *
+   * @return string
+   *   The URL.
+   */
+  public function getUrl() {
+    return $this->url;
+  }
+
+  /**
+   * Get the redirect status code.
+   *
+   * @return int
+   *   The status code.
+   */
+  public function getStatusCode() {
+    return $this->statusCode;
+  }
+
+  /**
+   * Get the redirect headers.
+   *
+   * @return \string[]
+   *   The array of header strings.
+   */
+  public function getHeaders() {
+    return $this->headers;
+  }
+
+  /**
+   * Get HTTP redirect response object.
+   *
+   * @return \Drupal\Core\Routing\TrustedRedirectResponse
+   *   The response object.
+   */
+  public function getHttpResponseObject() {
```

Comment:
If we rename the class, this method can be called just getResponse() or getRedirectResponse().